app/models/ordergroup.rb

```diff
@@ -54,9 +54,10 @@ def get_available_funds(exclude = nil)
 
   # Creates a new FinancialTransaction for this Ordergroup and updates the account_balance accordingly.
   # Throws an exception if it fails.
-  def add_financial_transaction!(amount, note, user)
-    transaction do      
-      t = FinancialTransaction.new(:ordergroup => self, :amount => amount, :note => note, :user => user)
+  def add_financial_transaction!(amount, note, user, transaction_type = nil)
+    transaction_type = FinancialTransactionType.default if transaction_type.nil?
+    transaction do
+      t = FinancialTransaction.new(:ordergroup => self, :amount => amount, :note => note, :user => user, :financial_transaction_type => transaction_type)
```

Comment:
What about moving the default transaction type to `FinancialTransaction` itself? In that way plugins creating a financial transaction using the constructor (I've done it at some point, but not sure if that survived or not in the final plugin), would still work. Perhaps an `after_initialize` there to set the default transaction, documenting that it is for backward compatibility.
What do you think?